pages/[detailPictureView].js

```diff
@@ -10,19 +10,43 @@ export default function DetailPictureViewPage({ countries, country }) {
   }
 
   const { publicIds, imagesUrls } = country;
-  const publicId = publicIds.findIndex((id) => id === currentPublicId);
+  const publicIdIndex = publicIds.findIndex((id) => id === currentPublicId);
 
-  const currentPicture = imagesUrls[publicId];
+  const currentPicture = imagesUrls[publicIdIndex];
```

Comment:
Cant you find the currentPicture in your imagesUrls array with `imagesUrls.find()` and search with the currentPublicId? If not, maybe you can update your imagesUrls array that they include the publicIds?

Reply:
I don't really get this...because I need the publicIds and the imagesUrls separat. I can't mix them together. During the upload of the picture the imageUrl and the publicId go to the same places in the arrays so imageUrl of picture 1 is at the first place in the imageUrls array and the publicId of picture 1 is at the first place of the publicIds array.  So I think it's a good way to work here with the index. Correct me if I'm wrong.

Reply:
I tried to write the code by using `imagesUrls.find()` but I think Kilian is right that he needs to find the index of the Id in publicIds in order to get the url. Maybe both could live together in an array of objects like `[{id: "xyz", url: "www.abc.de"}]` but im not really sure would be a big benefit? (depends how they are used in the code otherwise)

Reply:
hang on, maybe this would work? 
imagesUrls.find((url, index) => publicIds[index] === currentPublicId)

Reply:
Thank you for the options. But I think Felix doesn't like the index. But for me it's the only way to solve this...